packages/devtools/lib/src/timeline/timeline_protocol.dart

```diff
@@ -453,6 +453,21 @@ class TimelineEvent {
   bool get isCpuEventFlow => _hasChild('Engine::BeginFrame');
   bool get isGpuEventFlow => _hasChild('PipelineConsume');
 
+  /// Returns the depth of this TimelineEvent tree, including [this].
+  int get depth => _depth ?? _getDepth(this);
+  int _depth;
+
+  int _getDepth(TimelineEvent root) {
```

Comment:
nit: add a comment that we assume TimelineEvent nodes are not modified after getDepth is called the first time. If that was supported we would need to clear the cache.

Reply:
done.